### ISSUE TITLE
Android NDK 18b issues

### ISSUE THREAD

**Issue by tito** [kivy/python-for-android#1525]
Libffi latest stable version is incompatible with Android NDK 18b:
```
../src/arm/sysv.S:363:2: error: invalid instruction, did you mean: fldmiax?
 fldmiadgt ip, {d0-d7}
 ^
../src/arm/sysv.S:399:2: error: invalid instruction
 stmeqia r2, {r0, r1}
 ^
../src/arm/sysv.S:412:2: error: invalid instruction, did you mean: fstmiax?
 fstmiadeq r2, {d0-d3}
 ^
/tmp/sysv-5d4296.s:263:2: error: invalid instruction
 fstmfdd sp!, {d0-d7}
 ^
/tmp/sysv-5d4296.s:313:2: error: invalid instruction, did you mean: fldmiax?
 fldmiad sp, {d0-d1}
 ^
/tmp/sysv-5d4296.s:316:2: error: invalid instruction, did you mean: fldmiax?
 fldmiad sp, {d0-d3}
```

It appears to be because of clang 7.0, which doesn't accept deprecated functions. 
The Android libffi version have fixes: https://android.googlesource.com/platform/external/libffi/+/ca22c3cb49a8cca299828c5ffad6fcfa76fdfa77

Didn't tested if libffi master have the fixes, but stable don't.

**Comment by inclement** [comment]
I found this still happened with NDK 19 in #1612, so we'll need to deal with this it seems.

**Comment by dbrnz** [comment]
FYI, libffi can be made to compile with NDK 19 by using setting  `version = 3.3-rc0` and `patches = []` in the libffi recipe.

**Comment by inclement** [comment]
Thanks @dbrnz, that's great to know.

**Comment by misl6** [comment]
Fixed via #1722
